core/src/main/java/hudson/util/Iterators.java

```diff
@@ -316,7 +317,8 @@ public void remove() {
      */
     @SafeVarargs
     public static <T> Iterable<T> sequence( final Iterable<? extends T>... iterables ) {
-        return () -> new FlattenIterator<T,Iterable<? extends T>>(ImmutableList.copyOf(iterables)) {
+        return () -> new FlattenIterator<T,Iterable<? extends T>>(Collections.unmodifiableList(StreamSupport.stream(
+            Arrays.stream(iterables).spliterator(), true).collect( Collectors.toList()))) {
```

Comment:
Simpler would be to use the `Stream` object returned by `Arrays.stream` rather than wrapping it in another stream:

```suggestion
        return () -> new FlattenIterator<T,Iterable<? extends T>>(Collections.unmodifiableList(
            Arrays.stream(iterables).collect(Collectors.toList()))) {
```

Reply:
good catch